### PR TITLE
CI: HoTT doesn't use autogen anymore.

### PR DESCRIPTION
**Pull request by SkySkimmer**
Since https://github.com/HoTT/HoTT/pull/1476

### CODE REVIEW

File: dev/ci/ci-hott.sh [hunk 1]
```diff
@@ -5,6 +5,5 @@ ci_dir="$(dirname "$0")"
 
 git_download hott
 
-( cd "${CI_BUILD_DIR}/hott" && ./autogen.sh -skip-submodules && ./configure \
-      && sed -i.bak 's/\(HOQC =.*\)/\1 -native-compiler no/' Makefile \
-      && make && make validate )
+export COQEXTRAFLAGS='-native-compiler no'
```
Comment (JasonGross) [inline dev/ci/ci-hott.sh:8]: This line should not be needed anymore (we pass `-native-compiler ondemand`).

Comment (SkySkimmer) [reply dev/ci/ci-hott.sh:8]: I saw the pr removing it and figured we may as well be explicit.

Comment (JasonGross) [reply dev/ci/ci-hott.sh:8]: I've commented on that PR suggesting that we don't remove it and instead pass `-native-compiler no`.  I believe we'll be tracking this on our CI, too, and I'm hopeful that HoTT will work with the native compiler in the future, and we'll only remove it then when it makes sense to build with the native compiler.  That is, I don't think HoTT deserves any special treatment on Coq's CI here.

**Review by ejgallego — APPROVED** [review]
Seems fine.

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**